sklearn/calibration.py

```diff
@@ -257,7 +266,13 @@ def fit(self, X, y, sample_weight=None):
             else:
                 check_is_fitted(self.base_estimator)
             with suppress(AttributeError):
-                self.n_features_in_ = base_estimator.n_features_in_
+                if self.n_features_in_ != base_estimator.n_features_in_:
```

Comment:
This changes makes it so `CalibrationCV` does not delegate to the inner estimator, but will set `n_feature_in_` by itself. This is not inline with [SLEP010](https://scikit-learn-enhancement-proposals.readthedocs.io/en/latest/slep010/proposal.html#considerations):

> In most meta-estimators, the input validation is handled by the sub-estimator(s). The n_features_in_ attribute of the meta-estimator is thus explicitly set to that of the sub-estimator, either via a @property, or directly in fit().

Although, I would be okay with changing this requirement and having meta-estimators set `n_features_in_` itself and validate against the `base_estimator`.

Reply:
If we implement https://github.com/scikit-learn/scikit-learn/pull/19555#discussion_r584180075 we would not need to change the SLEP: we can just check for consistency for prefit base estimators while still delegating to the base estimators in general.